climada/engine/impact.py

```diff
@@ -1372,7 +1372,147 @@ def _selected_exposures_idx(self, coord_exp):
+        for attr_name in attributes:
+           if not all(hasattr(imp, attr_name) for imp in imp_list):
+               raise ValueError(f"Attribute {attr_name} is not shared by all impacts. "
+                                "The impacts are incompatible and cannot be concatenated.")
+
+        #check if impacts have same frequency values (only first entries are checked)
+        freqs=set([imp.frequency[0] for imp in imp_list])
+        if len(freqs)>1:
+           raise ValueError(f"Impacts do not have the same frequency: {freqs}."
+                            "The impacts are incompatible and cannot be concatenated.")
+        frequencies=[imp.frequency for imp in imp_list]
+        impact.frequency = np.concatenate(frequencies,axis=0) 
+       
+        ### The following checks are specific to the selected concatenation options
+       
+        if concat_type=='time_window':
+          
+          #check if no overlapping dates
+          dates=[date for imp in imp_list for date in imp.date]
+          
+          if len(dates)!=len(set(dates)): 
+              raise ValueError("There is at least one date in more than one impact. Please make sure impacts do not contain same dates."
+                               "The impacts are incompatible and cannot be concatenated.")
+          impact.date = np.array(dates)
+          
+          
+          #check if exposure is consistent
+          isin=[np.isin(imp.coord_exp,imp_list[0].coord_exp) for imp in imp_list[0::]]
+          tot_vals=set([imp.tot_value for imp in imp_list])
+          if (not np.all(isin)) or (len(tot_vals)>1):
+               raise ValueError("Impacts are not based on the same exposure. Please make sure exposures are identical."
+                                "The impacts are incompatible and cannot be concatenated.")
+          exp_descriptions= {imp.tag['exp'].description for imp in imp_list if imp.tag['exp'].description != ''}
+          if len(exp_descriptions) > 1:
+               raise ValueError(f"The given impacts are based on exposures with different descriptions: {exp_descriptions}. "
+                            "The impacts are incompatible and cannot be concatenated.")      
+          impact.coord_exp=imp_list[0].coord_exp
+          impact.tot_value=imp_list[0].tot_value
+          
+          #fill remaining attributes
+          impact.tag=imp_list[0].tag
+
+          #event IDs
+          event_ids=[event_id for imp in imp_list for event_id in imp.event_id]
+          if len(event_ids)!=len(set(event_ids)):
+              raise ValueError("Duplicate event IDs found.")
+          impact.event_id=np.array(event_ids)
+          
+          #event names
+          event_names=[event_name for imp in imp_list for event_name in imp.event_name]
+          if len(event_names)!=len(set(event_names)):
+              raise ValueError("Duplicate event names found.")   
+          impact.event_name = np.array(event_names,dtype=object)
```

Comment:
```suggestion
          event_name = event_names
```
Event name is a list.